jobs/sql/extractions/mch_diagnoses.sql

```diff
@@ -0,0 +1,287 @@
+create temporary table temp_mch_diagnoses
+(
+	patient_id  int,
+	encounter_id int,
+	encounter_location  varchar(255),
+	obs_id  int,
+	obs_group_id int,
+	obs_datetime datetime,
+	visit_id int,
+	encounter_type varchar(100),
+	entered_by varchar(255),
+	provider varchar(255),
+	diagnosis_concept int,
+	diagnosis_entered   text,
+	diagnosis_coded_fr	varchar(255),
+	icd10_code			varchar(255),
+	dx_order			varchar(255),
+	certainty			varchar(255),
+	coded				varchar(255),
+	retrospective		int(1),
+	date_created		datetime,
+	abortion bit,
+	abortion_with_sepsis bit,
+	anemia bit,
+	cervical_cancer bit,
+	cervical_laceration bit,
+	complete_abortion bit,
+	diabetes int,
+	dystocia bit,
+	eclampsia bit,
+	hemorrhage bit,
+	hypertension bit,
+	incomplete_abortion bit,
+	induced_abortion bit,
+	postpartum_hemorrhage bit,
+	laceration_of_perineum bit,
+	malaria bit,
+	postnatal_complication bit,
+	preeclampsia bit,
+	puerperal_infection bit,
+	spontaneous_abortion bit,
+	sti bit,
+	threatened_abortion bit
+);
+
+insert into temp_mch_diagnoses (
+patient_id,
+encounter_id,
+obs_id,
+obs_group_id,
+obs_datetime,
+encounter_type,
+date_created,
+diagnosis_concept,
+diagnosis_entered,
+diagnosis_coded_fr,
+coded,
+retrospective
+)
+select 
+person_id,
+encounter_id,
+obs_id,
+obs_group_id,
+obs_datetime,
+encounter_type_name(encounter_id),
+date_created,
+value_coded,
+if(o.value_coded is null, o.value_text, concept_name(o.value_coded,'en')),
+concept_name(o.value_coded,'fr'),
+if(o.value_coded is null, 0,1),
+if(time_to_sec(date_created) - time_to_sec(obs_datetime) > 1800,1,0)
+from obs o 
+where 
+concept_id in (concept_from_mapping('PIH','DIAGNOSIS'), concept_from_mapping('PIH','Diagnosis or problem, non-coded'))
+and obs_group_id in (select obs_id from obs o1 where voided = 0 and concept_id = concept_from_mapping('PIH','Visit Diagnoses'))
+and encounter_id in (select encounter_id from encounter e where voided = 0 and encounter_type = @mch_encounter)
+;
+
+update temp_mch_diagnoses tm set visit_id = (select visit_id from encounter e where e.voided = 0 and tm.encounter_id = e.encounter_id);
+
+-- Categories
+-- Abortion
+
+update temp_mch_diagnoses tm set abortion = if(diagnosis_concept in (concept_from_mapping('CIEL', '120295'), concept_from_mapping('PIH', '7697'), concept_from_mapping('CIEL', '162206')), 1, 0);
+update temp_mch_diagnoses tm set abortion_with_sepsis = if(diagnosis_concept = concept_from_mapping('CIEL', '150746'), 1, 0);
+update temp_mch_diagnoses tm set anemia = if(diagnosis_concept in (
+concept_from_mapping('CIEL', '149566'), 
+concept_from_mapping('PIH', '3703'), 
+concept_from_mapping('CIEL', '1226'),
+concept_from_mapping('CIEL', '137932'),
+concept_from_mapping('CIEL', '162042'), 
+concept_from_mapping('PIH', '9325'),
+concept_from_mapping('PIH', '9088'),
+concept_from_mapping('PIH', '9089'),
+concept_from_mapping('PIH', '3'),
+concept_from_mapping('PIH', 'ANEMIA OF PREGNANCY'),
+concept_from_mapping('PIH', '9431')
+), 1, 0);
+
```

Comment:
@jmbabazi I noticed these statements run kind of slow against HUM Reporting.  Can you try loading each of these concept ids into variables first and then in the update statement you just check against those variables?  I wonder if it is evaluating concept_from_mapping over and over again?   I'm not sure it will be quicker but it's worth checking.

Reply:
thanks @dmdesimone , I made the changes